File: Exercism/ruby/raindrops/raindrops.rb
# frozen_string_literal: true

# Raindrops ...
class Raindrops
  def self.convert(num)
    factors = get_factors(num)
    result = ''.dup

    result << 'Pling' if factors.include? 3
    result << 'Plang' if factors.include? 5
    result << 'Plong' if factors.include? 7
    result << num.to_s if result.empty?

    result
  end

  class << self
    def get_factors(num)
      (1..num + 1).select { |i| (num % i).zero? }
    end
  end
end

File: Exercism/fortran/raindrops/raindrops.f90
module raindrops
  implicit none
contains

  function convert(i)
    integer :: i
    character(20) :: convert
    convert = ""

    if (mod(i, 3) == 0) then
       convert = trim(convert) // "Pling"
    end if
    if (mod(i, 5) == 0) then
       convert = trim(convert) // "Plang"
    end if
    if (mod(i, 7) == 0) then
       convert = trim(convert) // "Plong"
    end if
    if (len_trim(convert) == 0) then
       write(convert, '(I10)') i
       convert = trim(adjustl(convert))
    end if
  end function convert

end module raindrops
